game/views.py

```
@@ -1248,26 +1253,219 @@ def post(self, request, *args, **kwargs):
+    acquired = False
+    # Spin lock: attempt to acquire for up to 5 seconds
+    for _ in range(100):
+        if cache.add(lock_key, 1, timeout=10):
+            acquired = True
+            break
+        time.sleep(0.05)
+
+    if not acquired:
+        raise RuntimeError(
+            "Failed to acquire cache lock for increment_counter."
+        )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)